x-pack/legacy/plugins/apm/server/lib/helpers/get_apm_indices.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Server } from 'hapi';
+import { isObject } from 'lodash';
+import { getSavedObjectsClient } from './saved_objects_client';
+
+interface UiIndices {
+  apm_oss: {
+    sourcemapIndices?: string;
+    errorIndices?: string;
+    onboardingIndices?: string;
+    spanIndices?: string;
+    transactionIndices?: string;
+    metricsIndices?: string;
+    apmAgentConfigurationIndex?: string;
+  };
+}
+
+export async function getApmIndices(server: Server) {
+  const config = server.config();
+
+  const savedObjectsClient = getSavedObjectsClient(server, 'data');
+  // eslint-disable-next-line @typescript-eslint/camelcase
+  let apm_oss: UiIndices['apm_oss'] = {};
+  try {
+    const apmUiIndices = await savedObjectsClient.get<UiIndices>(
+      'apm-ui-indices',
+      'apm-ui-indices'
+    );
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    apm_oss = apmUiIndices.attributes.apm_oss;
+  } catch (err) {
+    server.log('error', err);
+  }
+  return {
```

Comment:
Maybe we can return a more compact object for a better experience for consumers:

```typescript
{
  sourcemaps: apm_oss.sourcemapIndices || config.get<string>('apm_oss.sourcemapIndices')
}
```

Usage downstream would then look like:
```typescript
const { indices: { sourcemaps } } = config;
```

Reply:
The purpose of the nested object is to match the configuration name, so that it is very clear which config the particular index is set in `kibana.yml`. I've considered flattening it, but decided against it so there was no confusion for the consumer. I'm open to changing this, but that was my rationale behind the nested object.